docs/sql-ref-literals.md

```diff
@@ -275,15 +275,30 @@ E [ + | - ] digit [ ... ]
 
     Case insensitive, indicates `DECIMAL`, with the total number of digits as precision and the number of digits to right of decimal point as scale.
 
+* **default (no postfix, no exponent)**
```

Comment:
I think we should probably just comebine them into one. @yaooqinn do you know any other cases or variants?

Reply:
IMHO, this new section is not necessary
- Necessary `Parameters` are listed and explained above
- For `decimal/double literals`: their SYNTAXes already described these behaviors
 

Keeping the examples might be enough.

Reply:
FYI, an internal config will also change this behavior

Reply:
Understanding what the literal type is from the syntax is not clear at all. Maybe it's just me, but I spent a good minute or two looking at that first and not getting that `BD` is the default in the case of no exponent.

Now that I know that's true, I can see that this bit of syntax can be reduced as follows:

```
decimal_digits { [ BD ] | [ exponent BD ] } | digit [ ... ] [ exponent ] BD
decimal_digits { [ BD ] | [ exponent BD ] }
decimal_digits { [ BD ] }
decimal_digits
```

But I doubt most users will find this easy to follow just by looking at the syntax.

The new bullets explaining the default explicitly are way easier to understand, and also mirror a similar bullet for integral literals:

https://github.com/apache/spark/blob/b4346ceffab74cae4cc1d6e053d5bde3b5387d5f/docs/sql-ref-literals.md#L197-L199

Reply:
> FYI, an internal config will also change this behavior

Interesting. I assume we don't want to mention this since it's an internal config, correct?

Reply:
I see your point that the syntax technically explains this, though I personally don't find it that user-friendly.

I'm fine with keeping just the examples.

Reply:
Using a style, such as BNF, like most commercial database vendors do, helps reduce the ambiguity. Users who are unfamiliar with these syntaxes may need to take some time to learn them. It's good to give more instructions and examples without introducing extra maintenance effort, ambiguity, etc.

Reply:
> Interesting. I assume we don't want to mention this since it's an internal config, correct?

Yes, we rarely(maybe never) do that